### PR TITLE
fix(DTFS2-6002): TFM all facilities graphql query fix

### PR DESCRIPTION
**Pull request by Zainzzkk**
# Issue:
* All facilities endpoint for tfm-api (graphql) crashing when extra '-' in coverEndDate

# Changes made:
* Added regex check for coverEndDate to ensure covers right format of 'YYYY-MM-DD'

### CODE REVIEW

**Review by abhi-markan — APPROVED** [review]

File: trade-finance-manager-api/src/graphql/resolvers/query-facilities.js [hunk 1]
```diff
@@ -19,15 +19,23 @@ exports.queryAllFacilities = async (queryParams) => {
     // finds latest completed amendment tfm object with mapped values
     const latestCompletedAmendment = findLatestCompletedAmendment(item?.amendments);
 
+    let facilityCoverEndDate = '-';
+    let coverEndDateEpoch = '';
+    // regex to check date is correct format - YYYY-MM-DD
+    const dateRegex = /^(19|20)\d{2}-(0[1-9]|1[0-2])-(0[1-9]|1\d|2\d|3[01])$/;
```
Comment (abhi-markan) [inline trade-finance-manager-api/src/graphql/resolvers/query-facilities.js:25]: Can we simplify this please. Else looks good.

**Review by abhi-markan — CHANGES_REQUESTED** [review]

File: trade-finance-manager-api/src/graphql/resolvers/query-facilities.js (hunk 1 above)
Comment (Zainzzkk) [reply trade-finance-manager-api/src/graphql/resolvers/query-facilities.js]: done

File: trade-finance-manager-api/src/graphql/resolvers/query-facilities.js [hunk 2]
```diff
@@ -19,15 +19,23 @@ exports.queryAllFacilities = async (queryParams) => {
     // finds latest completed amendment tfm object with mapped values
     const latestCompletedAmendment = findLatestCompletedAmendment(item?.amendments);
 
+    let facilityCoverEndDate = '-';
```
Comment (abhi-markan) [inline trade-finance-manager-api/src/graphql/resolvers/query-facilities.js:22]: Is there a reason why the variable `facilityCoverEndDate` has been instantiated with `-`?

**Review by abhi-markan — CHANGES_REQUESTED** [review]
Do we need to add a test or is it already covered?

File: trade-finance-manager-api/src/graphql/resolvers/query-facilities.js (hunk 2 above)
Comment (Zainzzkk) [reply trade-finance-manager-api/src/graphql/resolvers/query-facilities.js:22]: fixed

**Review by abhi-markan — APPROVED** [review]

**Pull request merged**